pkg/distributor/query.go

```diff
@@ -226,10 +229,21 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, userID string, re
 					// We expect to be always able to convert the label matchers back to Prometheus ones.
 					// In case we fail (unexpected) the error will not include the matchers, but the core
 					// logic doesn't break.
-					matchers, _ := ingester_client.FromLabelMatchers(req.Matchers)
 					return nil, validation.LimitError(fmt.Sprintf(errMaxChunksPerQueryLimit, util.LabelMatchersToString(matchers), chunksLimit))
 				}
 			}
+			for _, series := range resp.Chunkseries {
+				limitErr := queryLimiter.AddSeries(series.Labels, matchers)
+				if limitErr != nil {
```

Comment:
[nit] You can compress this:

```suggestion
				if limitErr := queryLimiter.AddSeries(series.Labels, matchers); limitErr != nil {
```

_Same comment applies below._